AdobeStockAsset/Model/AssetRepository.php

```diff
@@ -96,16 +130,52 @@ public function __construct(
         $this->collectionProcessor = $collectionProcessor;
         $this->searchResultFactory = $searchResultFactory;
         $this->assetSaveService = $commandSave;
-        $this->assetKeywordRepository = $assetKeywordRepository;
+        $this->getAssetKeywords = $getAssetKeywords;
+        $this->saveAssetKeywords = $saveAssetKeywords;
+        $this->saveMediaAsset = $saveMediaAsset;
+        $this->saveAssetLinks = $saveAssetLinks;
+        $this->mediaAssetInterfaceFactory = $mediaAssetInterfaceFactory;
     }
 
     /**
-     * @inheritdoc
+     * Save asset
+     *
+     * @param AssetInterface $asset
+     * @return void
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
      */
     public function save(AssetInterface $asset): void
     {
+        $mediaGalleryAssetId = $this->saveMediaGalleryAsset($asset);
```

Comment:
Adobe Stock Asset Repository should be exclusively used to save Adobe Stock Asset. Please move the media gallery asset saving to `Magento\AdobeStockAsset\Model\SaveAsset` class